daffodil-runtime1/src/main/scala/org/apache/daffodil/layers/AISTransformer.scala

```diff
@@ -77,6 +78,11 @@ class AISPayloadArmoringTransformer()
     val newJOS = new LayerBoundaryMarkInsertingJavaOutputStream(jos, layerBoundaryMark, iso8859)
     newJOS
   }
+
+  def startLayerForParse(s: PState): Unit = () // do nothing
```

Comment:
These should all have "override" keyword on them. 

I believe so far we have only needed the start flavor for parse, and the end flavor for unparse, so end for parse, and start for unparse could be removed.